calyx/src/ir/guard.rs

```diff
@@ -52,13 +114,41 @@ impl Guard {
             Guard::Geq(_, _) => ">=".to_string(),
             Guard::Leq(_, _) => "<=".to_string(),
             Guard::Not(_) => "!".to_string(),
-            Guard::Port(_) => panic!("No operator string for Guard::Port"),
+            Guard::Port(_) | Guard::True => {
+                panic!("No operator string for Guard::Port")
+            }
         }
     }
 
-    ////////////// Convinience constructors ///////////////////
-    pub fn and(self, other: Guard) -> Self {
-        Guard::And(vec![self, other])
+    pub fn and_vec(&self, guards: &mut Vec<Guard>) -> Self {
+        flatten_and(guards);
+        let mut new;
+        if let Guard::And(inner) = &self {
+            new = inner.clone();
```

Comment:
These `clones` seem unnecessary. Either make the function take ownership of `self` or define it as an associated function.